openedx_external_enrollments/tests/external_enrollments/tests_viper_external_enrollment.py

```diff
@@ -142,3 +143,68 @@ def test_get_course_home_url_with_failed_post_enrollment(self, post_mock, model_
+    @patch('openedx_external_enrollments.external_enrollments.viper_external_enrollment.requests.post')
+    def test_refresh_api_keys_with_failed_post_request(self, post_mock, request_log_create_mock):
+        """
+        This test validates the _refresh_api_keys method with a failed post request.
+        """
+        module = 'openedx_external_enrollments.external_enrollments.viper_external_enrollment'
+        post_mock.side_effect = Exception('Exception reason.')
+
+        with LogCapture(level=logging.ERROR) as log_capture:
+            result = self.base._refresh_api_keys()  # pylint: disable=protected-access
+            log = '[viper\'s] API key refresh execution failed. Reason: Exception reason.'
+            log_capture.check((module, 'ERROR', log), )
+
+        request_log_create_mock.assert_called_once_with(
+            request_type=str(self.base),
+            details=log
+        )
+        self.assertEqual(result[0], 'Exception reason.')
+        self.assertEqual(result[1], status.HTTP_400_BAD_REQUEST)
+
+    @patch('openedx_external_enrollments.external_enrollments.viper_external_enrollment.EnrollmentRequestLog.objects.create')  # noqa: disable=E501 pylint: disable=C0301
+    @patch('openedx_external_enrollments.external_enrollments.viper_external_enrollment.requests.post')
+    def test_refresh_api_keys_with_success_post_request(self, post_mock, request_log_create_mock):
+        """
+        This test validates the _refresh_api_keys method with a successful post request.
+        """
+        module = 'openedx_external_enrollments.external_enrollments.viper_external_enrollment'
+        mock_response = Mock(status_code=status.HTTP_200_OK)
+        mock_response.json.return_value = {'message': 'succeeded'}
+        post_mock.return_value = mock_response
+        with LogCapture(level=logging.INFO) as log_capture:
+            result = self.base._refresh_api_keys()  # pylint: disable=protected-access
+            log = '[viper\'s] API key successfully refreshed -- {\'message\': \'succeeded\'}'
+            log_capture.check((module, 'INFO', log), )
+
+        request_log_create_mock.assert_called_once_with(
+            request_type=str(self.base),
+            details=log
+        )
+        self.assertEqual(result[0], log)
+        self.assertEqual(result[1], status.HTTP_200_OK)
+
+    @patch('openedx_external_enrollments.external_enrollments.viper_external_enrollment.EnrollmentRequestLog.objects.create')  # noqa: disable=E501 pylint: disable=C0301
+    @patch('openedx_external_enrollments.external_enrollments.viper_external_enrollment.requests.post')
+    def test_refresh_api_keys_with_success_post_request_and_bad_request(self, post_mock, request_log_create_mock):
+        """
+        This test validates the _refresh_api_keys method with a successful post request.
```

Comment:
@ivanvgh A part of the name test says `succes_post`, but for me it is a failed post, isn't it?. `post_mock` returns `mock_response` which is `Mock(status_code=status.HTTP_400_BAD_REQUEST)`
``
```suggestion
        This test validates the _refresh_api_keys method with a bad post request.
```